tests/unit/core/binarycodec/types/test_issue.py

```diff
@@ -0,0 +1,69 @@
+from unittest import TestCase
+
+from xrpl.core.binarycodec import XRPLBinaryCodecException
+from xrpl.core.binarycodec.binary_wrappers.binary_parser import BinaryParser
+from xrpl.core.binarycodec.types.issue import Issue
+
+
+class TestIssue(TestCase):
+    def test_from_value_xrp(self):
+        issue_obj = Issue.from_value({"currency": "XRP"})
+        self.assertEqual(issue_obj.to_json(), {"currency": "XRP"})
+
+    def test_from_value_issued_currency(self):
+        test_input = {"currency": "USD", "issuer": "rG1QQv2nh2gr7RCZ1P8YYcBUKCCN633jCn"}
+        issue_obj = Issue.from_value(test_input)
+        expected = {"currency": "USD", "issuer": "rG1QQv2nh2gr7RCZ1P8YYcBUKCCN633jCn"}
+        self.assertEqual(issue_obj.to_json(), expected)
+
+    def test_from_parser_xrp(self):
+        # Test round-trip: serialize an XRP Issue and then parse it back.
+        test_input = {"currency": "XRP"}
```

Comment:
Is this a valid input for the binary-codec ? `"currency": "<160-bit-representation-of-currency>"`

Can you include tests for non-standard currency codes? Reference: https://xrpl.org/docs/references/protocol/data-types/currency-formats#nonstandard-currency-codes

Reply:
Yes, it's a valid input; you can refer to AMM using this format. And I just added the tests for non-standard currency code.